### PR TITLE
feat(docs): add cheatseet for ML07

### PR DESCRIPTION
**Pull request by aryanxk02**
This PR adds cheatsheet for transfer learning attack. 
Ref: https://github.com/OWASP/www-project-machine-learning-security-top-10/issues/155
CC: @shsingh @techiemac

### CODE REVIEW

**Review by techiemac — CHANGES_REQUESTED** [review]
Thanks for the PR! I think overall this should come from the lens of transfer learning attack mitigations and not necessarily what transfer learning is.

File: Top10MLSummary.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+## ML01:2023 Input Manipulation Attack
```
Comment (techiemac) [inline Top10MLSummary.md:1]: For now, let's hold off on the summaries. I appreciate the ownership but work still needs to be done on the core docs. I think once that is complete, we will just lift the Description of each one into the respective summaries.

**Review by techiemac — CHANGES_REQUESTED** [review]

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Improve performance on new tasks using knowledge from related tasks.
+- Enable effective learning with limited labeled data by transferring knowledge from large datasets.
+
+#### Strategies
+1. **Feature Extraction**: Freeze early layers of pre-trained models and extract learned features for new tasks.
+2. **Fine-tuning**: Unfreeze some layers of pre-trained models and continue training with new data to adapt to specific tasks.
+3. **Domain Adaptation**: Adjust pre-trained models to new domains by transferring knowledge while minimizing domain shift.
+4. **Multi-task Learning**: Train models to perform multiple tasks simultaneously, leveraging shared representations for improved performance.
+
+#### Implementation
```
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:19]: Thanks for the review @techiemac. What mitigations here for transfer learning attack. And how much should be the content?

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+### Transfer Learning in Machine Learning Cheat Sheet
+
+#### Introduction
```
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:3]: Alright, got it!

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Transfer learning (TL) is a machine learning (ML) technique that uses a previously trained model as the base for building a new model for a different task. The model is trained using less data for the new task. TL is a popular approach in deep learning because it can train deep neural networks with less data than building a model from scratch.
+
+This cheat sheet provides concise guidance for implementing Transfer Learning in Machine Learning models via Keras and TensorFlow.
+
+#### Goals of Transfer Learning
+- Speed up training by initializing models with pre-learned parameters.
+- Improve performance on new tasks using knowledge from related tasks.
+- Enable effective learning with limited labeled data by transferring knowledge from large datasets.
+
+#### Strategies
```
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:13]: I'll research and add around 4-5 risks associated with transfer learning attack.

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:19]: And do I enhance the Strategies part or completely replace it with the mitigations?

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+1. **Feature Extraction**: Freeze early layers of pre-trained models and extract learned features for new tasks.
+2. **Fine-tuning**: Unfreeze some layers of pre-trained models and continue training with new data to adapt to specific tasks.
+3. **Domain Adaptation**: Adjust pre-trained models to new domains by transferring knowledge while minimizing domain shift.
+4. **Multi-task Learning**: Train models to perform multiple tasks simultaneously, leveraging shared representations for improved performance.
+
+#### Implementation
+```python
```
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:20]: Okay, I will have to go through the repo once and check what's the content in there :)

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+
+```python
+# Compile model
+model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
+
+# Train model on new data
+model.fit(train_data, train_labels, epochs=10, batch_size=32, validation_data=(val_data, val_labels))
+```
+### Best Practices
```
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:58]: Thanks, got it!

File: docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+**Regularization**: Prevent overfitting during fine-tuning with techniques like dropout or weight decay.
+
+**Transfer Subset of Layers**: Avoid overfitting or loss of information by transferring only relevant layers.
+
+**Monitor Performance**: Continuously monitor model performance and adjust training strategy accordingly.
+
+**Experiment with Architectures**: Explore different architectures and pre-trained models for best performance.
+
+### Conclusion
```
[the remark this answers is not in the document]
Comment (aryanxk02) [reply docs/cheatsheets/ML02_2023-Transfer-Learning-Attack-Cheatsheet.md:70]: Okay, I'll make the changes.
